src/live_restore/live_restore_fs.c

```diff
@@ -1261,21 +1303,54 @@ __live_restore_fs_open_file(WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const ch
                  * the file. We're bypassing the live_restore layer so we don't try to modify the
                  * extents in hole_list_head.
                  */
-                WT_ERR(lr_fh->destination.fh->fh_truncate(
-                  lr_fh->destination.fh, wt_session, source_size));
+                WT_RET(lr_fhp->destination.fh->fh_truncate(
+                  lr_fhp->destination.fh, wt_session, source_size));
 
                 /*
                  * Initialize the extent as one hole covering the entire file. We need to read
                  * everything from source.
                  */
-                WT_ASSERT(session, lr_fh->destination.hole_list_head == NULL);
-                WT_ERR(__live_restore_alloc_extent(
-                  session, 0, (size_t)source_size, NULL, &lr_fh->destination.hole_list_head));
+                WT_ASSERT(session, lr_fhp->destination.hole_list_head == NULL);
+                WT_RET(__live_restore_alloc_extent(
+                  session, 0, (size_t)source_size, NULL, &lr_fhp->destination.hole_list_head));
             }
         } else
-            lr_fh->destination.complete = true;
+            lr_fhp->destination.complete = true;
     }
 
+    return (0);
+}
+
+/*
+ * __live_restore_fs_open_file --
+ *     Open a live restore file handle. This will: - If the file exists in the source, open it in
+ *     both. - If it doesn't exist it'll only open it in the destination.
+ */
+static int
+__live_restore_fs_open_file(WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *name,
+  WT_FS_OPEN_FILE_TYPE file_type, uint32_t flags, WT_FILE_HANDLE **file_handlep)
+{
+    WT_DECL_RET;
+    WT_LIVE_RESTORE_FILE_HANDLE *lr_fh;
+    WT_LIVE_RESTORE_FS *lr_fs;
```

Comment:
If you want a lot of these declarations can happen later, I'm not fussed if you don't want to make that change though.

Reply:
[1fe24c5](https://github.com/wiredtiger/wiredtiger/pull/11379/commits/1fe24c5407d375cef1de5a0536e4ee4aa57eee51)